### PR TITLE
Mechanism for adding/removing "tags" to/from tensor names, after changes to the tensor

### LINKED ISSUE #454: Implement a mechanism for adding/removing "tags" to/from tensor names, after changes to the tensor
🐞 **What bugs you?**

Every Tensor Object has a `name` parameter, that stores the user-defined (and modifiable) name. After any change is made to the tensor, for example a Lorentz Transform or a Config Change, the name remains unmodified. This may make tracking changes to the tensor difficult for extended calculations.

🎯 **Goal**

A tag added to the Tensor name, after **binary** changes are made to the Tensor to mark the respective changes, would be nice to have. For example, after change_config or lorentz_transformation, tags, like "_cc" or "_lt" can be appended/removed.

💡 **Possible solutions**

Write a helper function for altering `name` conditionally.

Edit `BaseRelativityTensor.lorentz_transform()`.

Edit `change_config()` for all Tensors in `einsteinpy.symbolic`.

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by JeS24**
Fixes #454 

@ritzvik @shreyasbapat Sorry for the other PR. I could not rebase properly. Please review this PR.

### CODE REVIEW

File: src/einsteinpy/symbolic/tensor.py [hunk 1]
```diff
@@ -245,8 +249,8 @@ def __init__(
         functions : tuple or list or set
             List of symbolic functions used in epressing the tensor. 
             Calculates in real-time if left blank.
-        name : string
-            Name of the Tensor. Defaults to None.
+        name : str
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/tensor.py]:
```suggestion
        name : str or None
```

**Comment by ritzvik** [comment]
If possible add a test for `_change_name` here

https://github.com/einsteinpy/einsteinpy/blob/master/src/einsteinpy/tests/test_symbolic/test_helpers.py

**Comment by ritzvik** [comment]
> If possible add a test for `_change_name` here
> 
> https://github.com/einsteinpy/einsteinpy/blob/master/src/einsteinpy/tests/test_symbolic/test_helpers.py

test for both string parameter and `None`type

**Comment by JeS24** [comment]
@ritzvik Done.

**Comment by ritzvik** [comment]
For quality checks
https://docs.einsteinpy.org/en/latest/dev_guide.html#code-linting

**Comment by JeS24** [comment]
> For quality checks
> https://docs.einsteinpy.org/en/latest/dev_guide.html#code-linting

I keep forgetting to run `tox` before committing. 😞

**Pull request merged**